### PR TITLE
ARUHA-843 Be able to not specify startup timeout

### PR DESCRIPTION
**Pull request by antban**
At the same time - be able to detect process startup fail

### CODE REVIEW

File: bubuku/broker.py [hunk 1]
```diff
@@ -160,20 +173,16 @@ def _is_leadership_transferred(self, active_broker_ids=None, dead_broker_ids=Non
...
                     if any(str(x) in active_broker_ids for x in state.get('isr', [])):
-                        _LOG.warn(
+                        _LOG.warning(
                             'Leadership is not transferred for {} {} ({}, brokers: {})'.format(
                                 topic, partition, json.dumps(state), active_broker_ids))
                         return False
                     else:
-                        _LOG.warn('Shit happens! No single isr available for {}, {}, state: {}, '
-                                  'skipping check for that'.format(topic, partition, json.dumps(state)))
+                        _LOG.warning('Shit happens! No single isr available for {}, {}, state: {}, '
```
Comment (adyach) [inline bubuku/broker.py]: do not use dirty words in logs

File: bubuku/broker.py [hunk 2]
```diff
@@ -125,14 +132,20 @@ def start_kafka_process(self, zookeeper_address):
...
-            self.process_holder.set(self._open_process())
-
-            _LOG.info('Waiting for kafka to start up with timeout {} seconds'.format(self.timeout.get_timeout()))
-            if not self.id_manager.wait_for_broker_id_presence(self.timeout.get_timeout()):
-                self.timeout.on_timeout_fail()
+            self.process.start(self.kafka_properties.settings_file)
+
+            _LOG.info('Waiting for kafka to start with timeout {}'.format(self.timeout))
+            start = time()
+            while self.process.is_running():
```
Comment (adyach) [inline bubuku/broker.py]: maybe not running ?

File: bubuku/daemon.py [hunk 3]
```diff
@@ -77,7 +78,7 @@ def main():
 
     config = load_config()
     _LOG.info("Using configuration: {}".format(config))
-    process_holder = KafkaProcessHolder()
+    process_holder = KafkaProcess(config.kafka_dir)
```
Comment (adyach) [inline bubuku/daemon.py]: process ?

File: bubuku/broker.py (hunk 2 above)
Comment (antban) [inline bubuku/broker.py]: No, here it is fine

File: bubuku/daemon.py (hunk 3 above)
Comment (antban) [inline bubuku/daemon.py]: Done https://github.com/zalando-incubator/bubuku/pull/97/commits/a62853bdf0b96e3d95d457c469a193564dd6113d

File: bubuku/broker.py (hunk 1 above)
Comment (antban) [inline bubuku/broker.py]: @adyach Fixed https://github.com/zalando-incubator/bubuku/pull/97/commits/a62853bdf0b96e3d95d457c469a193564dd6113d

**Comment by adyach** [comment]
👍

**Comment by antban** [comment]
:+1:

**Comment by adyach** [comment]
Merged other waiting tickets for validation in that one.

**Comment by adyach** [comment]
👍

**Comment by adyach** [comment]
👍

**Comment by adyach** [comment]
👍

**Comment by adyach** [comment]
👍

**Comment by adyach** [comment]
👍

**Comment by adyach** [comment]
👍

**Pull request merged**